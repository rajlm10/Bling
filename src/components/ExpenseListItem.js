import React from 'react'
import { Link } from 'react-router-dom'

const ExpenseListItem=(props)=>{
    return(
        <div>
            <h3><Link to={`/edit/${props.expense.id}`}>{props.expense.description}</Link></h3>
            <p>{props.expense.amount} - {props.expense.createdAt}</p>

        </div>
    )
}

export default (ExpenseListItem)