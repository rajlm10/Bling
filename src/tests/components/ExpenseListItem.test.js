import React from 'react'
import {shallow} from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import moment from 'moment'

const expenses=[{
    id:1,
    description:'Gum',
    note:'',
    amount:195,
    createdAt:0
},
{
    id:2,
    description:'Rent',
    note:'',
    amount:109500,
    createdAt:moment(0).subtract(4,'days').valueOf()
},
{
    id:3,
    description:'Credit Card',
    note:'',
    amount:4500,
    createdAt:moment(0).add(4,'days').valueOf()
}]

test('should render expense ',()=>{
    const wrapper=shallow(<ExpenseListItem key={expenses[0].id} expense={expenses[0]}/>)
    expect(wrapper).toMatchSnapshot()
})