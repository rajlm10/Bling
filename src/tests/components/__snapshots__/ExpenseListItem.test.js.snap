// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense  1`] = `
<div>
  <h3>
    <[object Object]
      to="/edit/1"
    >
      Gum
    </[object Object]>
  </h3>
  <p>
    195
     - 
    0
  </p>
</div>
`;
